spec/system/create_tokens_spec.rb

```diff
@@ -10,37 +10,16 @@
 
     visit campaign_path(map.campaign, as: user)
     wait_for_connection
-    find(".new-token-link").click
+    click_on "New Token"
     within "[data-controller=modal]" do
       fill_in "Name", with: "Olokas"
       attach_file "Image", file_fixture("olokas.jpeg")
-      click_on "Create Token"
+      click_on "Create"
     end
 
+    # if we don't use find here, capybara doesn't wait for the ajax to complete
+    html_token = find(".token[data-target='map.token']")
```

Comment:
Would changing the matcher to use `#find` solve this?

Reply:
No, I see what you're doing. The new token's id isn't available until the request is performed.